src/riak_core_handoff_receiver.erl

```diff
@@ -122,15 +126,21 @@ process_message(?PT_MSG_INIT, MsgData, State=#state{vnode_mod=VNodeMod,
     State#state{partition=Partition, vnode=VNode};
 
 process_message(?PT_MSG_BATCH, MsgData, State) ->
-    lists:foldl(fun(Obj, StateAcc) -> process_message(?PT_MSG_OBJ, Obj, StateAcc) end, 
+    lists:foldl(fun(Obj, StateAcc) -> process_message(?PT_MSG_OBJ, Obj, StateAcc) end,
                 State,
                 binary_to_term(MsgData));
 
-process_message(?PT_MSG_OBJ, MsgData, State=#state{vnode=VNode, count=Count}) ->
+process_message(?PT_MSG_OBJ, MsgData, State=#state{vnode=VNode, count=Count,
+                                                   vnode_timeout_len=VNodeTimeout}) ->
     Msg = {handoff_data, MsgData},
-    case gen_fsm:sync_send_all_state_event(VNode, Msg, 60000) of
+    case catch gen_fsm:sync_send_all_state_event(VNode, Msg, VNodeTimeout) of
```

Comment:
Here's a stab at it, note you don't have to handle exceptions you don't care about.

```erlang
    try gen_fsm:sync_send_all_state_event(VNode, Msg, VNodeTimeout) of
        ok ->
            State#state{count=Count+1};
        E={error,_} ->
            exit(E)
    catch
        exit:{timeout,_} ->
            exit({error, {vnode_timeout, VNodeTimeout, size(MsgData),
                          binary:part(MsgData, {0,min(size(MsgData),128)})}})
    end;
```